examples/dft/compile_time_dispatching/CMakeLists.txt

```diff
@@ -19,25 +19,45 @@
 
 #Build object from all sources
 set(DFTI_CT_SOURCES "")
+
 if(ENABLE_MKLGPU_BACKEND)
   list(APPEND DFTI_CT_SOURCES "complex_fwd_buffer_mklgpu")
 endif()
 
+if(ENABLE_MKLCPU_BACKEND)
+  list(APPEND DFTI_CT_SOURCES "complex_fwd_buffer_mklcpu")
+endif()
+
+include(WarningsUtils)
+
 foreach(dfti_ct_sources ${DFTI_CT_SOURCES})
-  add_executable(example_${domain}_${dfti_ct_sources} ${dfti_ct_sources}.cpp)
+  # add executable and define include directories
+  # add dependencies and link libraries
+  # register example as ctest
+ add_executable(example_${domain}_${dfti_ct_sources} ${dfti_ct_sources}.cpp)
   target_include_directories(example_${domain}_${dfti_ct_sources}
       PUBLIC ${PROJECT_SOURCE_DIR}/examples/include
       PUBLIC ${PROJECT_SOURCE_DIR}/include
       PUBLIC ${CMAKE_BINARY_DIR}/bin
   )
-  if(domain STREQUAL "dft" AND ENABLE_MKLGPU_BACKEND)
-    add_dependencies(example_${domain}_${dfti_ct_sources} onemkl_${domain}_mklgpu)
-    list(APPEND ONEMKL_LIBRARIES_${domain} onemkl_${domain}_mklgpu)
+
+  set(ONEMKL_LIBRARIES_${domain} "")
+  if(domain STREQUAL "dft")
```

Comment:
is domain ever not dft?
That applies to a lot of this file.
Otherwise it looks a lot better :)